src/solution.jl

```diff
@@ -34,7 +34,11 @@ function DataDrivenSolution(b::AbstractBasis, p::AbstractDataDrivenProblem,
     ps = get_parameter_values(b)
     prob = remake_problem(p, p = ps)
 
-    rss = sum(abs2, get_implicit_data(prob) .- b(prob))
+    if size(b(prob)) == size(get_implicit_data(prob))
+        rss = sum(abs2, get_implicit_data(prob) .- b(prob))
+    else
+        rss = sum(abs2, get_implicit_data(prob))
```

Comment:
how is this the residual?

Reply:
I ran into the DimensionMismatch error while using DataDrivenSparse.jl. It seemed to occur when the basis is empty. I assumed that this would be analog to computing the residual to a vector of zeros. When there is no model to approximate the data, the residual can be seen as the data itself.